conda-store image specification fix

viniciusdc opened the pull request:
closes #309 #590 

Creating a new image/dockerfile for [conda-store](https://github.com/Quansight/qhub/tree/main/qhub/template/%7B%7B%20cookiecutter.repo_directory%20%7D%7D/image)  to QHub. This should allow QHub to deploy custom conda-store images as well as solve some libraries missing tools from the current conda-store.

costrouc commented:
This part of the PR looks good @viniciusdc. Now in this same PR could you add the ability to specify the conda-store image being used? An example of this would be to look at the `default_images.dask-gateway` key.

viniciusdc commented:
Hi @costrouc , could you have a look at the new additions (I think I addressed everything). For this part: 
the `default_images.conda-store` will be fed into the conda-store module https://github.com/Quansight/qhub/blob/51ed41bd1cc1bb13cb0e3fdb729ef6837c1ffcd3/qhub/template/%7B%7B%20cookiecutter.repo_directory%20%7D%7D/infrastructure/kubernetes.tf#L87
 I'm not sure how to proceed.

viniciusdc commented:
Hi @costrouc, do you have any ideas on why Terraform is complaining about main.tf configuration?

viniciusdc commented:
I think I may have found the issue, maybe cookiecutter is not providing the references to the image name correctly? 
```python 
[terraform]:     "containers": [
[terraform]:      {
[terraform]:       "name": "conda-store",
[terraform]:       "image": "{{ cookiecutter.default_images.conda_store.split(':')[0] }}:{{ cookiecutter.default_images.conda_store.split(':')[1] }}",
[terraform]:       "command": [
```

viniciusdc commented:
It seems to be working (I also tested the specification of the sha image tag in `qhub-config.yml`) 
 ```shell 
(base) root@qhub-conda-store-7754899784-cv4v9:/# apt list | grep mesa

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

libd3dadapter9-mesa-dev/stable 18.3.6-2+deb10u1 amd64
libd3dadapter9-mesa/stable 18.3.6-2+deb10u1 amd64
libegl-mesa0/stable,now 18.3.6-2+deb10u1 amd64 [installed,automatic]
libegl1-mesa-dev/stable 18.3.6-2+deb10u1 amd64
libegl1-mesa/stable,now 18.3.6-2+deb10u1 amd64 [installed]
libgl1-mesa-dev/stable 18.3.6-2+deb10u1 amd64
libgl1-mesa-dri/stable,now 18.3.6-2+deb10u1 amd64 [installed,automatic]
libgl1-mesa-glx/stable,now 18.3.6-2+deb10u1 amd64 [installed]
libglapi-mesa/stable,now 18.3.6-2+deb10u1 amd64 [installed,automatic]
libgles2-mesa-dev/stable 18.3.6-2+deb10u1 amd64
libgles2-mesa/stable 18.3.6-2+deb10u1 amd64
libglu1-mesa-dev/stable 9.0.0-2.1+b3 amd64
libglu1-mesa/stable 9.0.0-2.1+b3 amd64
libglw1-mesa-dev/stable 8.0.0-1.1+b1 amd64
libglw1-mesa/stable 8.0.0-1.1+b1 amd64
libglx-mesa0/stable,now 18.3.6-2+deb10u1 amd64 [installed,automatic]
libosmesa6-dev/stable 18.3.6-2+deb10u1 amd64
libosmesa6/stable 18.3.6-2+deb10u1 amd64
libwayland-egl1-mesa/stable 18.3.6-2+deb10u1 amd64
mesa-common-dev/stable 18.3.6-2+deb10u1 amd64
mesa-opencl-icd/stable 18.3.6-2+deb10u1 amd64
mesa-utils-extra/stable 8.4.0-1+b1 amd64
mesa-utils/stable 8.4.0-1+b1 amd64
mesa-va-drivers/stable 18.3.6-2+deb10u1 amd64
mesa-vdpau-drivers/stable 18.3.6-2+deb10u1 amd64
mesa-vulkan-drivers/stable 18.3.6-2+deb10u1 amd64
```

costrouc commented on qhub/initialize.py:
```diff
@@ -40,6 +40,7 @@
         "jupyterlab": f"quansight/qhub-jupyterlab:{qhub_image_tag}",
         "dask_worker": f"quansight/qhub-dask-worker:{qhub_image_tag}",
         "dask_gateway": f"quansight/qhub-dask-gateway:{qhub_image_tag}",
+        "conda_store": "quansight/qhub-conda-store:sha-a24165f",
```
Should be `quansight/qhub-conda-store:{qhub_image_tag}`

viniciusdc replied:
Yes I know, I only did that because the CI tests were failing.

costrouc reviewed (changes requested):
Fix naming for default_image.conda_store

brl0 commented on qhub/template/{{ cookiecutter.repo_directory }}/image/Dockerfile.conda-store:
```diff
@@ -0,0 +1,15 @@
+FROM quansight/conda-store:e2051a36e60bd3abd9aa44105f240b359ee6bab7
+
+RUN apt-get update \
```
Just a thought, but might be worth considering adding the apt package `build-essential`, it includes a couple of compilers, make, and development libraries that are somewhat commonly needed when building packages.

costrouc reviewed (approved):
Looks great to me! Thanks @viniciusdc
